cpp/tests/c_api/random_walks_test.c

```diff
@@ -0,0 +1,434 @@
+
+  ret_code = cugraph_make_device_buffer(p_handle, vertex_tid, num_edges, &dbuf_src);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "src device_buffer creation failed.");
+
+  ret_code = cugraph_make_device_buffer(p_handle, vertex_tid, num_edges, &dbuf_dst);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "dst device_buffer creation failed.");
+
+  ret_code = cugraph_make_device_buffer(p_handle, weight_tid, num_edges, &dbuf_wgt);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer creation failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_src, (byte_t*)h_src);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "src device_buffer update failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_dst, (byte_t*)h_dst);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "dst device_buffer update failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, weight_tid, &dbuf_wgt, (byte_t*)h_wgt);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer update failed.");
+
+  p_sampling_strategy = cugraph_create_sampling_strategy(0, 0.0, 0.0);
+  runtime_assert(p_sampling_strategy != NULL, "sampling strategy creation failed.");
+
+  p_graph_envelope = cugraph_make_sg_graph(p_handle,
+                                           vertex_tid,
+                                           edge_tid,
+                                           weight_tid,
+                                           FALSE,
+                                           &dbuf_src,
+                                           &dbuf_dst,
+                                           &dbuf_wgt,
+                                           num_vertices,
+                                           num_edges,
+                                           FALSE,
+                                           FALSE,
+                                           FALSE);
+  runtime_assert(p_graph_envelope != NULL, "graph envelope creation failed.");
+
+  ret_code    = cugraph_random_walks(p_handle,
+                                  p_graph_envelope,
+                                  &dbuf_start,
+                                  num_paths,
+                                  max_depth,
+                                  flag_use_padding,
+                                  p_sampling_strategy,
+                                  &rw_ret);
+  flag_failed = (ret_code != CUGRAPH_SUCCESS);
+  runtime_assert(flag_failed == TRUE, "cugraph_random_walks() should have failed.");
+
+  cugraph_free_rw_result(&rw_ret);
+
+  cugraph_free_graph(p_graph_envelope);
+
+  cugraph_free_sampling_strategy(p_sampling_strategy);
+
+  cugraph_free_device_buffer(&dbuf_start);
+
+  cugraph_free_device_buffer(&dbuf_wgt);
+
+  cugraph_free_device_buffer(&dbuf_dst);
+
+  cugraph_free_device_buffer(&dbuf_src);
+
+  cugraph_free_handle(p_handle);
+
+  /* because bools and err return codes from programs have opposite meaning...*/
+  if (flag_failed)
```

Comment:
I think this is still inverted: tests need to return 0 on success, but this will return 0 if flag_failed is true, meaning it's a failure? Maybe I'm confused...

edit: oh, this flag is only used for the check above which must "fail". I would possibly rename the variable, or add something to the comment, or change the name of the test, or all of the above.

Reply:
It's correct. This is a negative test.